scripts/specializations/CpCourseManager.lua

```diff
@@ -204,14 +206,17 @@ function CpCourseManager:setCoursesFromNetworkEvent(courses)
 end
 
 function CpCourseManager:addCourse(course,noEventSend)
-    local spec = self.spec_cpCourseManager 
+    local spec = self.spec_cpCourseManager
+    -- reset temporary offset field course, this will be regenerated based on the current settings when the job starts
+    spec.offsetFieldWorkCourse = nil
     course:setVehicle(self)
     table.insert(spec.courses,course)
     SpecializationUtil.raiseEvent(self,"onCpCourseChange",course,noEventSend)
 end
 
 function CpCourseManager:resetCourses()
-    local spec = self.spec_cpCourseManager 
+    local spec = self.spec_cpCourseManager
+    spec.offsetFieldWorkCourse = nil
```

Comment:
These resets are only applied, when the course changes. This might be a problem, if the user want to start the course with different lane offsets. For example the wrong lane offset was used and then restarted with a different lane offset.

Reply:
That is taken care of elsewhere, that is why we save the offsetFieldWorkPosition.

Reply:
Seems like I over looked this: "position ~= previousPosition" check. Might be okay then, but a test with AD and refill would still be usefull.